src/explorer/CosmosDBTreeItem.ts

```diff
@@ -55,13 +55,24 @@ export class CosmosDBTreeItem extends AzureParentTreeItem<ISiteTreeRoot> {
         const treeItems: CosmosDBConnection[] = [];
         const usedLabels: { [key: string]: boolean } = {};
         for (const key of mongoAppSettingsKeys) {
-            const cosmosDBDatabase = await ext.cosmosAPI.getDatabase({ connectionString: appSettings[key] });
-            if (cosmosDBDatabase) {
-                const label = CosmosDBConnection.makeLabel(cosmosDBDatabase);
-                if (!usedLabels[label]) {
-                    usedLabels[label] = true;
-                    treeItems.push(new CosmosDBConnection(this, cosmosDBDatabase, key));
+            try {
+                const databaseTreeItem = await ext.cosmosAPI.findTreeItem({ connectionString: appSettings[key] });
```

Comment:
Please use `createTreeItemsWithErrorHandling`, which handles errors in a common way

Here's an example: https://github.com/Microsoft/vscode-azureappservice/blob/master/src/explorer/WebAppProvider.ts#L48